src/test/scala/io/iohk/ethereum/blockchain/sync/PivotBlockSelectorSpec.scala

```diff
@@ -110,6 +110,52 @@ class PivotBlockSelectorSpec
     )
   }
 
+  it should "retry if there are no enough eligible peers" in new TestSetup {
+    updateHandshakedPeers(
+      HandshakedPeers(
+        threeAcceptedPeers
+          .updated(peer1, allPeers(peer1).copy(maxBlockNumber = bestBlock - syncConfig.pivotBlockOffset - 1))
+      )
+    )
+
+    pivotBlockSelector ! SelectPivotBlock
+
+    peerMessageBus.expectNoMessage()
+
+    updateHandshakedPeers(HandshakedPeers(threeAcceptedPeers))
+
+    time.advance(syncConfig.startRetryInterval)
+
+    peerMessageBus.expectMsgAllOf(
+      Subscribe(MessageClassifier(Set(Codes.BlockHeadersCode), PeerSelector.WithId(peer1.id))),
+      Subscribe(MessageClassifier(Set(Codes.BlockHeadersCode), PeerSelector.WithId(peer2.id))),
+      Subscribe(MessageClassifier(Set(Codes.BlockHeadersCode), PeerSelector.WithId(peer3.id)))
+    )
+  }
```

Comment:
I could use
// given
// when
// then
and possibly comments on two assertions that are done in this test, it is unclear for me how retry manifest